# Halteria genome research
## Bioinformatics tools used in Halteria genome research.  
## Usage of all python (or perl) scripts:  
### (1) Code S1. simulation_binomial_sampling.py  
This is a python script.  
**Required module:** No additional python module is required.  
**Usage:** python simulation_binomial_sampling.py \<input file: table recording pool population data\> \<output file\> \<total reads number\>   
  **Eg.** python simulation_biomial_sampling.py pol_data.tab output_data.tab 186978547  
  
The input file is a file in table format. Format is shown below.  

|chromosome|position|minor_allele|MAF|major_allele|MAAF|position_type|poly_type|reads_num|gene|
|---|---|---|---|---|---|---|---|---|---|
|NODE_5302_length_4220_cov_357.479|30|T|0.264044943820225|A|0.69873595505618|inter_genic|bi|1424|NULL|  

### (2) Code S2. ADFinder_singleT.py  
This is a unpublished software. For detailed information, please see this URL "https://github.com/weibozheng/ADFinder"  

### (3) Code S3. blast_xml_protein_coding_sequence_extract_v2.pl  
This is a perl script.  
**Required module:** Bio::SeqIO, Bio::SearchIO. They are in the bioperl bundle (version 1.6).  
**Usage:** perl blast_xml_protein_coding_sequence_extract_v2.pl \<contig file: genome assembly\> \<XML file: genome assembly blastx nr or swiss prot database, generated by blastx software (ncbiblast++) using -outfmt 5\> \<output folder\>   
  **Eg.** perl blast_xml_protein_coding_sequence_extract_v2.pl genome.fa genome_blastx_swiss.xml output_folder  
  
### (4) Code S4. blast_xml_protein_coding_sequence_extract_v2.pl  
This is a perl script.  
**Required module:** Bio::SeqIO, Bio::SearchIO. They are in the bioperl bundle (version 1.6).  
**Usage:** perl blast_xml_protein_coding_sequence_extract_v2.pl \<contig file: genome assembly\> \<XML file: genome assembly blastx nr or swiss prot database, generated by blastx software (ncbiblast++) using -outfmt 5\> \<output folder\>   
  **Eg.** perl blast_xml_protein_coding_sequence_extract_v2.pl genome.fa genome_blastx_swiss.xml output_folder  
